Userland/Libraries/LibWeb/CSS/Percentage.h

```diff
@@ -231,38 +231,53 @@ template<>
 struct AK::Formatter<Web::CSS::Percentage> : Formatter<StringView> {
     ErrorOr<void> format(FormatBuilder& builder, Web::CSS::Percentage const& percentage)
     {
-        return Formatter<StringView>::format(builder, percentage.to_deprecated_string());
+        auto maybe_string = percentage.to_string();
+        if (maybe_string.is_error())
+            return Formatter<StringView>::format(builder, "[out of memory]"sv);
+        return Formatter<StringView>::format(builder, maybe_string.release_value());
     }
 };
 
 template<>
 struct AK::Formatter<Web::CSS::AnglePercentage> : Formatter<StringView> {
     ErrorOr<void> format(FormatBuilder& builder, Web::CSS::AnglePercentage const& angle_percentage)
     {
-        return Formatter<StringView>::format(builder, angle_percentage.to_deprecated_string());
+        auto maybe_string = angle_percentage.to_string();
+        if (maybe_string.is_error())
+            return Formatter<StringView>::format(builder, "[out of memory]"sv);
+        return Formatter<StringView>::format(builder, maybe_string.release_value());
     }
 };
 
 template<>
 struct AK::Formatter<Web::CSS::FrequencyPercentage> : Formatter<StringView> {
     ErrorOr<void> format(FormatBuilder& builder, Web::CSS::FrequencyPercentage const& frequency_percentage)
     {
-        return Formatter<StringView>::format(builder, frequency_percentage.to_deprecated_string());
+        auto maybe_string = frequency_percentage.to_string();
+        if (maybe_string.is_error())
+            return Formatter<StringView>::format(builder, "[out of memory]"sv);
+        return Formatter<StringView>::format(builder, maybe_string.release_value());
     }
 };
 
 template<>
 struct AK::Formatter<Web::CSS::LengthPercentage> : Formatter<StringView> {
     ErrorOr<void> format(FormatBuilder& builder, Web::CSS::LengthPercentage const& length_percentage)
     {
-        return Formatter<StringView>::format(builder, length_percentage.to_deprecated_string());
+        auto maybe_string = length_percentage.to_string();
+        if (maybe_string.is_error())
+            return Formatter<StringView>::format(builder, "[out of memory]"sv);
+        return Formatter<StringView>::format(builder, maybe_string.release_value());
     }
 };
 
 template<>
 struct AK::Formatter<Web::CSS::TimePercentage> : Formatter<StringView> {
     ErrorOr<void> format(FormatBuilder& builder, Web::CSS::TimePercentage const& time_percentage)
     {
-        return Formatter<StringView>::format(builder, time_percentage.to_deprecated_string());
+        auto maybe_string = time_percentage.to_string();
+        if (maybe_string.is_error())
+            return Formatter<StringView>::format(builder, "[out of memory]"sv);
+        return Formatter<StringView>::format(builder, maybe_string.release_value());
```

Comment:
There's already a formatter for ErrorOr<value>s that shows the value in braces if present, or the error. 
e.g. printing an `ErrorOr { String("Foo") } ` would show `"{foo}"`. I wonder if it's worth just adjusting this so that if the value is an `AK::String` the braces are not added, since printing a newly created string is quite a common thing and you don't necessarily want to error check that every time.

Reply:
Alternatively I think you could just do:
```cpp
return Formatter<StringView>::format(TRY(builder, time_percentage.to_string()));
```
Since `format()` already returns ErrorOr<> so we can just propagate the error.

Reply:
Thanks for the review! I made the changes and it's way better like this.

I also fixed some instances where I was unnecessarily converting the `to_string()` back to `to_deprecated_string()` in `dbgln` functions. Now it's just `to_string()`.